python/reg_interface/scripts/sca.py

```diff
@@ -47,68 +115,85 @@ def main():
-            printRed('file-type can be "mcs" or "bit"')
-            return
-
-        ftype = sys.argv[4]
-        filename = sys.argv[5]
-        program_fpga(ohMask,ftype,filename)
-
-    elif instructions == 'test1':
-        test1()
-        
-    elif instructions == 'test2':
-        test2()
-
-    elif instructions == 'compare-mcs-bit':
-        if len(sys.argv) < 5:
-            print("Usage: sca.py local compare-mcs-bit <mcs_filename> <bit_filename>")
-            return
-        mcsFilename = sys.argv[3]
-        bitFilename = sys.argv[4]
-        compare_mcs_bit(mcsFilename, bitFilename)
-
-    elif instructions == 'gpio-set-direction':
-        if len(sys.argv) < 4:
-            print('Usage: sca.py local gpio-set-direction <direction-mask>')
-            print('direction-mask is a 32 bit number where each bit represents a GPIO channel -- if a given bit is high it means that this GPIO channel will be set to OUTPUT mode, and otherwise it will be set to INPUT mode')
-            return
-        directionMask = parseInt(sys.argv[3])
-        gpio.set_direction(ohMask,directionMask)
-
-    elif instructions == 'gpio-set-output':
-        if len(sys.argv) < 4:
-            print('Usage: sca.py local gpio-set-output <output-data>')
-            print('output-data is a 32 bit number representing the 32 GPIO channels state')
-            return
-        outputData = parseInt(sys.argv[3])
-        gpio.set_output(ohMask,outputData)
-
-    elif instructions == 'gpio-read-input':
-        gpio.read_input(ohMask)
+    writeReg(getNode('GEM_AMC.SLOW_CONTROL.SCA.MANUAL_CONTROL.LINK_ENABLE_MASK'), args.ohMask)
+
+    return ohList
+
+def scaReset(args):
+    ohList = scaInit(args)
+    sca_reset(ohList)
+    return
+
+def sysmon(args):
+    scaInit(args)
+    run_sysmon(args.ohMask)
+    return
  
 if __name__ == '__main__':
-    main()
+    # create the parser
+    import argparse
+    parser = argparse.ArgumentParser(description='Arguments to supply to sca.py')
+
+    # Positional arguments
+    parser.add_argument("cardName", type=str, help="hostname of the AMC you are connecting too, e.g. 'eagle64'; if running on an AMC use 'local' instead", metavar="cardName")
+    parser.add_argument("ohMask", type=parseInt, help="ohMask to apply, a 1 in the n^th bit indicates the n^th OH should be considered", metavar="ohMask")
+    subparserCmds = parser.add_subparsers(help="sca command help")
+
+    # Create subparser for compare-mcs-bit
+    parser_compareFwFiles = subparserCmds.add_parser("compare-mcs-bit", help="compares a *.mcs with a *.bit file to check if the firmware is the same")
+    parser_compareFwFiles.add_argument("--fwFileBit",type=str, dest="fwFileBit", required=True,
+            help="firmware bit file to be used in the comparison", metavar="fwFileBit")
+    parser_compareFwFiles.add_argument("--fwFileMCS",type=str, dest="fwFileMCS", required=True,
+            help="firmware bit file to be used in the comparison", metavar="fwFileMCS")
```

Comment:
typo

Reply:
Can a comparison be done between two `bit` files?  Between two `mcs` files?

Reply:
The function that is called here is `compare_mcs_bit` which I guess is authored by @evka85.  It is specifically looking at a `bit` file compared against an `mcs` file.  So presently no a comparison between two `bit` files or two `mcs` files is not possible.

Reply:
> typo

Again could you be more specific?

Reply:
re: typos
line 147 `bit` was written in the help for the `mcs` file